src/views/ApartmentApp/components/StudentApplication/components/HallChoiceList/index.jsx

```diff
@@ -46,7 +47,21 @@ const HallChoiceList = ({
     };
 
     loadApartmentHalls();
-  }, []);
+
+    if (apartmentChoices?.length >= 5) {
+      setClickCount(5);
+    } else {
+      setClickCount(apartmentChoices?.length);
+    }
+  }, [apartmentChoices]);
+
+  const handleAddHallClick = () => {
+    // Only continue if click count is less than 5
+    if (clickCount < 5) {
+      onHallAdd();
+      setClickCount((prevClickCount) => prevClickCount + 1);
+    }
```

Comment:
```suggestion
    if (apartmentChoices?.length >= 5) {
      setClickCount(5);
    } else {
      setClickCount(apartmentChoices?.length);
    }
  }, [apartmentChoices]);

  const handleAddHallClick = () => {
    // Only continue if click count is less than 5
    if (clickCount < 5) {
      onHallAdd();
      setClickCount((prevClickCount) => prevClickCount + 1);
    }
```
I do not suggest using a click counter. This can easily be done by checking the number of components you're rendering in the hall list.